src/tools/rbd/action/Import.cc

```diff
@@ -400,6 +418,8 @@ int do_import_diff_fd(librados::Rados &rados, librbd::Image &image, int fd,
   r = (r < 0) ? r : temp_r; // preserve original error
   if (r == 0 && tosnap.length()) {
     idiffctx.image->snap_create(tosnap.c_str());
+    if (is_protected)
```

Comment:
Nit: Sorry just noticed that the error code isn't propagated. Can you modify the ```snap_create``` line above to store the return value in ```r```, change this to ```if (r == 0 && is_protected)```, and change the line below to ```r = ...snap_protect```?